src/include/stir/recon_buildblock/KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData.h

```diff
@@ -0,0 +1,471 @@
+
+    This file is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    See STIR/LICENSE.txt for details
+*/
+/*!
+  \file
+  \ingroup GeneralisedObjectiveFunction
+  \brief Declaration of class stir::KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData
+  \author Daniel Deidda
+
+
+*/
+
+#ifndef __stir_recon_buildblock_KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData_H__
+#define __stir_recon_buildblock_KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData_H__
+
+#include "stir/RegisteredParsingObject.h"
+#include "stir/recon_buildblock/PoissonLogLikelihoodWithLinearModelForMean.h"
+//#include "stir/ProjData.h"
+#include "stir/recon_buildblock/ProjectorByBinPair.h"
+//#include "stir/recon_buildblock/BinNormalisation.h"
+#include "stir/TimeFrameDefinitions.h"
+#ifdef STIR_MPI
+#include "stir/recon_buildblock/distributable.h" // for  RPC_process_related_viewgrams_type
+#endif
+#include "stir/Array.h"
+//#include "stir/DiscretisedDensity.h"
+#include "stir/shared_ptr.h"
+#include <string>
+//#include "stir/recon_buildblock/IterativeReconstruction.h"
+START_NAMESPACE_STIR
+
+class DistributedCachingInformation;
+
+//#ifdef STIR_MPI_CLASS_DEFINITION
+//#define KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData_MPI
+//#endif
+
+
+/*!
+  \ingroup GeneralisedObjectiveFunction
+  \brief An objective function class appropriate for PET emission data
+
+  Measured data is given by a ProjData object, and the linear operations
+  necessary for computing the gradient of the objective function
+  are performed via a ProjectorByBinPair object together with
+  a BinNormalisation object.
+
+  \see PoissonLogLikelihoodWithLinearModelForMean.
+
+  This class implements the objective function obtained using the Kernel method (KEM) and Hybrid kernel method (HKEM).
+  This implementation corresponds to the one presented by Deidda D et al, ``Hybrid PET-MR list-mode kernelized expectation
+  maximization reconstruction for quantitative PET images of the carotid arteries", IEEE MIC Atlanta, 2017. However, this is
+  the sinogram-based obgective function. Each voxel value of the image, \f$ \boldsymbol{\lambda}\f$, can be represented as a
+  linear combination using the kernel method. % If we have an image with prior information, we can construct for each voxel
+  \f$ j \f$ of the PET image a feature vector, $\f \boldsymbol{v}_j \f$, using the prior information. The voxel value,
+  \f$\lambda_j\f$, can then be described using the kernel matrix
+
+
+
+  \f[
+   \lambda_j=  \sum_{l=1}^L \alpha_l k_{jl}
+  \f]
+
+  where \f$k_{jl}\f$ is the \f$jl^{th}\f$ kernel element of the matrix, \f$\boldsymbol{K}\f$.
+  The resulting algorithm is the following:
```

Comment:
thanks for the extensive documentation. However, this line and stuff below is inaccurate. You are talking about MLEM or OSEM versions of HKEM. However, you are implementing the objective function here. it could be used with OSSPS presumably (or other algorithms).

Either adjust the formulas, or say that for OSEM, the algorithm would be...